### ISSUE TITLE
Can you make CHC's custom JS pleaceholders useable on other plugins that supports PAPI?

### ISSUE THREAD

**Issue by anar4732** [kangarko/ChatControl-Pro#1870]
for exemple i can use AbstractMenus'variables anywhere that supports PAPI.

thanks.

**Comment by radiant-ai** [comment]
AFAIK PAPI already has a JS addon which allows to create custom placeholders.

**Comment by anar4732** [comment]
> AFAIK PAPI already has a JS addon which allows to create custom placeholders.

PAPI not supports color codes like {#ffff00}

**Comment by anar4732** [comment]
Nope, PAPI has not globlal JS pleaceholders support. in PAPI need to copy the JS code everywhere using JS pleaceholder.

**Comment by anar4732** [comment]
Java.of("net.md_5.bungee.api.ChatColor").of("#CCCCCC")
this is not working even.

**Comment by kangarko** [comment]
Sorry, it is Java.type not Java.of

Please see https://winterbe.com/posts/2014/04/05/java8-nashorn-tutorial/ for a quick tutorial on running JavaScript.

Unfortunately I will not be able to assist you with creating JS variables. I understand PAPI lacks this feature, we may add it later but at presents our roadmap is already full.

**Comment by anar4732** [comment]
> No unfortunately this is out of scope of the ChatControl plugin and PAPI already has this functionality.
> 
> I am pretty sure you can use the Rhino JavaScript engine to call Java.of("net.md_5.bungee.api.ChatColor").of("#CCCCCC") or something like that to get the color working on 1.16
hi thank you for help.
is it for just rhino or works with nashorn engine too?

**Comment by kangarko** [comment]
not sure, you'd have to test :)
